bootstrap/kubeadm/api/v1beta1/kubeadm_types.go

```diff
@@ -382,6 +398,12 @@ type LocalEtcd struct {
 	// +optional
 	ExtraArgs map[string]string `json:"extraArgs,omitempty"`
 
+	// ExtraEnvs is an extra set of environment variables to pass to the control plane component.
+	// Environment variables passed using ExtraEnvs will override any existing environment variables, or *_proxy environment variables that kubeadm adds by default.
+	// This option takes effect only on Kubernetes >=1.31.0.
+	// +optional
+	ExtraEnvs []EnvVar `json:"extraEnvs,omitempty"`
```

Comment:
```suggestion
	ExtraEnvs []corev1.EnvVar `json:"extraEnvs,omitempty"`
```

Do we need the extra container?

Reply:
Could question. It aligns to what kubeadm did. Not entirely sure why

(cc @neolit123)

Reply:
i actually don't remember why. will try to dig out comments.
seems there is no need for that.

should we change it pre-release 1.31?

Reply:
oh, that was a wild problem related to defaulting using internal APIs
https://github.com/kubernetes/kubernetes/pull/120561
https://github.com/kubernetes/kubernetes/pull/120561/files#diff-769d744db9aca1b60fb0f12e0c34fea8523740dccd24db5f9315b284ce1b1fefR211
https://github.com/kubernetes/kubeadm/issues/2927

TL;DR a new custom wrapper struct was needed so that we can have a customer defaulter and not end up with defaulters that import internal k/k/pkg packages and causing weird behavior.

Reply:
Having the same struct that kubeadm uses really helps to do comparisons cross versions (and we have many of them!)

So if there are no objections, I will keep this aligned